x-pack/plugins/cases/common/api/cases/case.ts

```diff
@@ -87,24 +87,21 @@ const CaseBasicRt = rt.type({
   owner: rt.string,
 });
 
-const CaseExternalServiceBasicRt = rt.type({
-  connector_id: rt.string,
+/**
+ * The external service fields. Exporting here for use in the service transformation code so I can define
+ * a type without the connector_id field.
+ */
+export const CaseExternalServiceBasicRt = rt.type({
+  connector_id: rt.union([rt.string, rt.null]),
```

Comment:
`connector_id` will be `null` if the reference cannot be found with deserializing from the saved object.

Reply:
As long as the user cannot set the `connector_id` I think it's fine to have it `null`.